api/src/main/java/com/messagebird/MessageBirdClient.java

```diff
@@ -1767,4 +1771,54 @@ public void cancelNumber(String number) throws UnauthorizedException, GeneralExc
         final String url = String.format("%s/phone-numbers", NUMBERS_CALLS_BASE_URL);
         messageBirdService.deleteByID(url, number);
     }
+
+    /**
+     * Uploads a file and returns the assigned ID.
+     *
+     * @param binary the bytes of the file to upload.
+     * @param contentType the content type of the file (e.g. "image/png").
+     * @return FileUploadResponse
+     * @throws GeneralException      general exception
+     * @throws UnauthorizedException if client is unauthorized
+     * @see #downloadFile
+     */
+    public FileUploadResponse uploadFile(byte[] binary, String contentType) throws GeneralException, UnauthorizedException {
+        if (binary == null) {
+            throw new IllegalArgumentException("File binary must be specified.");
+        }
+        if (contentType == null) {
+            throw new IllegalArgumentException("Content type must be specified.");
+        }
+
+        final String url = MESSAGING_BASE_URL + FILES_PATH;
+        final Map<String, String> headers = new HashMap<>();
+        headers.put("Content-Type", contentType);
```

Comment:
can you add a filename field too which is an optional field in the header?

Reply:
https://github.com/messagebird/java-rest-api/pull/148/commits/a8807c0737fd17db6f9e9a34393e96b0589df225